Form/update availabilities

jduong09 opened the pull request:
## Feature
- Allows user to choose a time or days options, and allows user to choose a start date and end date for a range of meeting times to select.
- Also allows the user to narrow the table by start and end times, for better readability of table.

jduong09 committed:
Added inputs and selects for startDate, endDate, startTime, and endTime

jduong09 committed:
Changed useEffect dependencies.

jduong09 committed:
Finished functionality for dynamically showing table dates based on startDate and endDate inputs.

jduong09 committed:
Added ranges for start time and end time.

jduong09 committed:
Added new keys to pollSchema, functionality for updating poll should be working.

jduong09 committed:
Fixed patch route, removed comments.

ziggysauce reviewed (changes requested):
This should say 12:00 am

<img width="1385" alt="Screen Shot 2024-01-01 at 2 14 08 PM" src="https://github.com/jduong09/doodle/assets/25065329/bcda8d74-778b-4525-8d10-7f360baefc36">

---

I select a range of dates (Jan 11 - Feb 11) but it still shows dates outside of that range and lets me chose dates outside of that range

---

This name input probs doesn't need to be this long. And then the submit button can be inline with the input instead of beneath it

<img width="1424" alt="Screen Shot 2024-01-01 at 2 15 33 PM" src="https://github.com/jduong09/doodle/assets/25065329/4ea834d3-6db9-48f1-ab8e-9d80f30ff639">

jduong09 committed:
Fixed incorrect option html tag in endTime select tag. Fixed bug with nextWeek logic.

jduong09 commented:
## Update
- Fixed incorrect option tag displaying 12:00 P.M. instead of 12:00 A.M. at the end of the end time options tag.
- Fixed bug that displayed end date twice in table.

jduong09 committed:
moved lines of code.

jduong09 commented:
## Update
- Merging pull request into css/Version1 after agreement that this is working.
